From 50f5768ff5e8874e6a7cd4c105c20d766d1c8653 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:51:11 -0400
Subject: [iommu] vt-d: Finally enable translation for non-PCI devices

Message-id: <20140905015111.31297.13555.stgit@gir.stowe>
Patchwork-id: 91347
O-Subject: [RHEL7.1 PATCH 54/65] iommu/vt-d: Finally enable translation for non-PCI devices
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 66077edc972c1c8dc2cf08e96a956c2db9bd705c

commit 66077edc972c1c8dc2cf08e96a956c2db9bd705c
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Mon Mar 10 20:04:11 2014 -0700

    iommu/vt-d: Finally enable translation for non-PCI devices

    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 0a8e166..a180f10 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -2907,9 +2907,6 @@ static int iommu_no_mapping(struct device *dev)
 {
  int found;
 
- if (unlikely(!dev_is_pci(dev)))
-  return 1;
-
  if (iommu_dummy(dev))
   return 1;
 
-- 
1.7.1